### PR TITLE
Disable pointing for range = -1

### LINKED ISSUE #6651: Disable pointing for tools
I request to be able to completely disable pointing for tools (including the hand).

To implement this, I request to simply interpret the value `range=0` as a special value which disables pointing.
Currently, this almost works, but when you are inside a node, you still point it.
It seems to be impossible to force the player to be completely unable to point something. There are certain uses for this (see below).
There's also a bug: If you set `range=0` for the hand, it does not work at all. The range remains at the default 4 if you have the hand selected, but the range is 0 if you have an item on hand (e.g. cobblestone).

Rationale:
This is an easy and reliable way to completely disable rightclicking, mining and punching any nodes, entities or whatever.

Use cases:
- Prevent punching or mining anything while the player is in a bed (using a hand overwrite)
- Add a player status which temporarily disables hand pointing
- Add a teleporter beacon tool which, when you leftclick, saves the position on which you stand. For tools like this, pointing wouldn't make much sense

### LINKED ISSUE #10355: Allow disabling pointing altogether
## Problem

Pointing can not be disabled for tools: `range = 0` or `range = nan` result in pointing still working inside nodes

## Solutions

`range = nan` or `range = false` should disable pointing

## Additional context

Sometimes one wants to disable pointing for i. e. magazines

### PR DESCRIPTION
**Pull request by appgurueu**
Disables pointing for items with range = -1. Closes #10355 . Ready for Review.

## How to test

```lua
minetest.register_tool("testmod:rangetester", { description = "Range tester", range = -1 })
```

EDIT by paramat:
Also closes #6651

### CODE REVIEW

**Comment by nerzhul** [comment]
why not just any value < 0 ? it will be closer to many return codes we have in UNIX calls

File: src/client/game.cpp [hunk 1]
```diff
@@ -3070,10 +3070,9 @@ void Game::processPlayerInteraction(f32 dtime, bool show_hud, bool show_debug)
 
 #endif
 
-	PointedThing pointed = updatePointedThing(shootline,
-			selected_def.liquids_pointable,
-			!runData.btn_down_for_dig,
-			camera_offset);
+	PointedThing pointed;
+	if (d != -1) pointed = updatePointedThing(shootline, selected_def.liquids_pointable,
+			!runData.down_for_dig, camera_offset);
```
Comment (pyrollo) [inline src/client/game.cpp:3075]:
```suggestion
	if (d != -1)
		pointed = updatePointedThing(shootline,
				selected_def.liquids_pointable,
				!runData.btn_down_for_dig,
				camera_offset);
```
Compilation error due to `down_for_dig` not being a member of `GameRunData`
And code style

**Review by pyrollo — CHANGES_REQUESTED** [review]
Agree with nerzhul, any negative value could disable pointing (but -1 is also ok for me).

File: src/client/game.cpp [hunk 2]
```diff
@@ -3052,7 +3052,7 @@ void Game::processPlayerInteraction(f32 dtime, bool show_hud, bool show_debug)
 	case CAMERA_MODE_THIRD_FRONT:
 		shootline.start = camera->getHeadPosition();
 		// prevent player pointing anything in front-view
-		d = 0;
+		d = d == -1 ? d : 0;
```
Comment (HybridDog) [inline src/client/game.cpp:3055]: Pointing should be disabled here (according to the comment), so I think this should be `d = -1;`.

**Comment by paramat** [comment]
@appgurueu changes needed.

**Comment by appgurueu** [comment]
I'm doubting whether this is the way to go. I would like to maintain compatibility as much as possible. If -1 was used, older clients would have max hand range, which is not intended. Modders want to use +/- 0. In order to differentiate between "allow pointing if inside [very rare use case]" and "never allow pointing" we could use the float sign bit.

**Pull request closed without merging**